File: static/CWE-295/java/vulnerable.java
// CWE-295: Improper Certificate Validation
// A custom X509TrustManager accepts all certificates, disabling TLS validation.
// Vulnerable sink: empty checkServerTrusted() in custom X509TrustManager

import java.security.cert.X509Certificate;
import javax.net.ssl.SSLContext;
import javax.net.ssl.TrustManager;
import javax.net.ssl.X509TrustManager;

class CWE_295 {
    public void trustAll() throws Exception {
        TrustManager[] trustAll = new TrustManager[] {
            new X509TrustManager() {
                public void checkClientTrusted(X509Certificate[] chain, String authType) {}
                public void checkServerTrusted(X509Certificate[] chain, String authType) {}
                public X509Certificate[] getAcceptedIssuers() { return new X509Certificate[0]; }
            }
        };
        SSLContext sc = SSLContext.getInstance("TLS");
        sc.init(null, trustAll, new java.security.SecureRandom());
    }
}

File: static/CWE-295/python/vulnerable.py
# CWE-295: Improper Certificate Validation
# TLS certificate verification is disabled when making an HTTPS request.
# Vulnerable sink: requests.get(url, verify=False)

import requests
from flask import Flask, request

app = Flask(__name__)


@app.route("/fetch")
def fetch():
    url = request.args.get("url")
    response = requests.get(url, verify=False)
    return response.text
